### ISSUE TITLE
Walletstore command line app UX

### ISSUE THREAD

**Issue by sachayves** [status-im/nim-beacon-chain#1517]
Suggested improvements to the interactive command line app

>**Note:** to enter the interactive CLI run 
> ```
> build/beacon_node wallets create
> ```

```
Please back up the seed phrase now to a safe location as if you are protecting a
sensitive password. The seed phrase can be used to withdrawl funds from your
wallet.
```

 Is this the only way you can withdraw funds from your wallet? if this is the case, i would make it explicit. if it isn't, ignore. There's also a minor typo: withdrawl --> withdraw.

```
Did you back up your seed recovery phrase?
(please type 'yes' to continue or press enter to quit)
```

I can see users rushing through and clicking `enter` by mistake. suggestion: press `q` to quit instead of `enter`

```
FAT 2020-08-17 16:24:11.177+02:00 Unable to create wallet                    topics="beacnde" tid=74900 file=beacon_node.nim:1426 err="aborted wallet creation"
```
The above looks like an error message, but it's actually expected behaviour on quit. Would replace with a friendlier message. Something like:

> Looks like you've just quit the wallet creation process! If this was a mistake, please run `wallets create` again.

```
For your convenience, the wallet can be identified with a name of your choice.
Please enter a wallet name below or press ENTER to continue with a
machine-generated name.
```

On pressing enter to continue with a machine generated name, one sees the following

```
Wallet name: 
INF 2020-08-17 16:19:41.989+02:00 Wallet file written                        tid=71914 file=keystore_management.nim:404 path="/Users/hackingresearch/Library/Application Support/Nimbus/BeaconNode/wallets/c0ff3c24-bff4-3b42-864e-45f785703d60.json"
```

In other words, no wallet name is displayed.  I think it would be more intuitive here if the machine-generated name were displayed.

**Comment by zah** [comment]
@sachayves, I've tried to incorporate your suggestions in the recent rework of the wallet creation process. Would you review it again to potentially close this issue?

**Comment by sachayves** [comment]
Amazing! Will test it out in a couple of hours

**Comment by sachayves** [comment]
I think it's better. However, still have a couple of comments:

- I think typing `c` for continue should be enough (after all, you can type `q` for quit)

- I suspect a lot of people will get fed up of trying to generate a high-entropy password -- it took me quite a few tries..). Folks are also more likely to forget a specially crafted password. Is there any way we can allow lower-entropy passwords, or is it an audit requirement?

- If you get your seed phrase wrong three times, you exit without a message. From the user's perspective it's unclear whether they got it right the third time around, or whether the process ends because they got it wrong again. Would exit with something like: "Sorry, since you weren't able to correctly recall your seed phrase, we weren't able to create your wallet, please try again".

- Minor typo: `In case you've saved the **seek** phrase`

- Minor grammatical point: `separated by a` rather than `separated with a`

- `we strongly advise` rather than `we strongly advice`

- I know it's a pain because it requires a special case, but would be good to have it read `1 more attempt`, rather than `1 more attempts`.

**Comment by zah** [comment]
> I think typing c for continue should be enough (after all, you can type q for quit)

This asymmetry between going forward and quitting was intentional. My reasoning was that the risk of quitting is much lower than the risk of going forward without fully grasping the message, so this is reflected in the effort required from the user. Having that said, I don't have strong opinions about this and I'm willing to change it if more people weigh it or we have good evidence from UX research that it doesn't matter or leads to problems.

> I suspect a lot of people will get fed up of trying to generate a high-entropy password

This was indeed a suggestion from our audit. You can consider it an educational feature - it's probably not reasonable to trust the knowledge of users regarding what is considered a strong password nowadays. We can perhaps tweak the minimum entropy score a bit as the current value was selected rather arbitrarily.

> If you get your seed phrase wrong three times, you exit without a message.

Great point. I'll fix that together with the other grammatical issues.

**Comment by sachayves** [comment]
>  My reasoning was that the risk of quitting is much lower than the risk of going forward without fully grasping the message, so this is reflected in the effort required from the user

I see where you're going here, but I still disagree quite strongly. My intuition is that it doesn't make much of a difference to the grasp they have on the message. Since typing `c` is not an automatic reflex, you can safely assume that if they've successfully type `c` then they've read the message (this would not be the case if we were asking them to press `enter`). From my experience, typing `continue` didn't make me re-read the message, it was simply inconvenient (I suspect this will be true for most users).

> We can perhaps tweak the minimum entropy score a bit as the current value was selected rather arbitrarily.

This would be great. As low as possible while satisfying audit requirements?

> Great point. I'll fix that together with the other grammatical issues.

Sweet :)
